patches/net/minecraft/world/level/block/entity/BlockEntityType.java.patch

```diff
@@ -1,14 +1,24 @@
 --- a/net/minecraft/world/level/block/entity/BlockEntityType.java
 +++ b/net/minecraft/world/level/block/entity/BlockEntityType.java
-@@ -301,6 +_,11 @@
-         return this.validBlocks.contains(p_155263_.getBlock());
-     }
+@@ -268,6 +_,7 @@
+     );
+     public static final BlockEntityType<VaultBlockEntity> VAULT = register("vault", BlockEntityType.Builder.of(VaultBlockEntity::new, Blocks.VAULT));
+     private final BlockEntityType.BlockEntitySupplier<? extends T> factory;
++    // Neo: This field will be modified by BlockEntityTypeAddBlocksEvent event. Please use the event to add to this field for vanilla or other mod's BlockEntityTypes.
+     private final Set<Block> validBlocks;
+     private final Type<?> dataType;
+     private final Holder.Reference<BlockEntityType<?>> builtInRegistryHolder = BuiltInRegistries.BLOCK_ENTITY_TYPE.createIntrusiveHolder(this);
+@@ -299,6 +_,13 @@
  
-+    // Neo: Add block getter
-+    public Set<Block> getValidBlocks() {
-+        return java.util.Collections.unmodifiableSet(this.validBlocks);
+     public boolean isValid(BlockState p_155263_) {
+         return this.validBlocks.contains(p_155263_.getBlock());
 +    }
 +
++    /**
++     * Neo: Add getter for an immutable view of the set of valid blocks.
++     */
++    public Set<Block> getValidBlocks() {
++        return java.util.Collections.unmodifiableSet(this.validBlocks);
+     }
+ 
```

Comment:
Weird change in patches.
Could you check this?

Reply:
The neo source file and patch file for this looks fine to me
![image](https://github.com/neoforged/NeoForge/assets/40846040/ca8be9f8-ff89-477c-adc1-a755ddf67675)
![image](https://github.com/neoforged/NeoForge/assets/40846040/e09c2585-0926-4c46-89a3-2c7e6f57f243)

Reply:
The patch file seems to indicate that the getValidBlocks method was before isValid before your PR

Reply:
moved now

Reply:
No, the getter was always after `isValid()`. DiffPatch just decided that instead of adding a closing brace for the added method, it prefers using the closing brace of the preceding method for the added one and patching in a new closing brace for the preceding method. My guess is that this happened due to the new hunk above this one.

Reply:
This is what it looks like now. I don't think it matters if it is above or below. Will keep it as is

![image](https://github.com/neoforged/NeoForge/assets/40846040/df22bfab-6258-48d1-9df8-91c7c571de88)